sqlmesh/core/plan/stages.py

```diff
@@ -364,6 +363,19 @@ def build(self, plan: EvaluatablePlan) -> t.List[PlanStage]:
             )
         )
 
+        if snapshots_with_schema_migration:
+            missing_ids_before = {s.snapshot_id for s in missing_intervals_before_promote}
+            missing_ids_after = {s.snapshot_id for s in missing_intervals_after_promote}
+            # Only skip migrate for VIEWs that are already scheduled for recreation in Backfill.
```

Comment:
No, let's not do this. The reason is that some snapshots may rely on view schema for their own migrations. So we absolutely must migrate those views. Additionally, there's already logic in place to skip view recreation during evaluation when appropriate.